src/main/java/com/aws/greengrass/deployment/activator/DefaultActivator.java

```diff
@@ -136,13 +139,20 @@ void rollback(DeploymentDocument deploymentDocument, CompletableFuture<Deploymen
         }
 
         try {
-            Set<GreengrassService> servicesToTrackForRollback = rollbackManager.servicesToTrack();
             // Don't track services if they were already broken before the rolled back deployment, because they'd
             // be expected to still be broken
-            servicesToTrackForRollback.removeIf((s) ->
-                    rollbackManager.getAlreadyBrokenServices().contains(s.getName()));
-            logger.atDebug(MERGE_CONFIG_EVENT_KEY)
+            Set<GreengrassService> servicesToTrackForRollback = rollbackManager.servicesToTrack();
+
+            // Also don't track services if they have (transitive) hard dependencies on already-broken services
+            Set<GreengrassService> brokenServiceAndDependers
+                    = findServiceDependers(servicesToTrackForRollback, rollbackManager.getAlreadyBrokenServices());
+
+            servicesToTrackForRollback.removeAll(brokenServiceAndDependers);
+
+            logger.atInfo(MERGE_CONFIG_EVENT_KEY)
                     .kv("previouslyBrokenServices", rollbackManager.getAlreadyBrokenServices())
+                    .kv("brokenServiceDependers", brokenServiceAndDependers.stream().map(GreengrassService::getName)
```

Comment:
nit: key is misspelled - "brokenServiceAndDependers"

Reply:
Updated